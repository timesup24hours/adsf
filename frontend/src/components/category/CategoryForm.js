import React, { Component } from 'react'

export class CategoryForm extends Component {

  render() {
    // const { categories, department, category, types } = this.props.menu
    return (
      <div className='CategoryForm-container'>
        <div className="row">
          <div className="col s12">
            <div className="row">
              <div className="input-field col s6">
                <input placeholder="Department" id="department" type="text" className="validate" />
                <label className='active' htmlFor="department">Department</label>
              </div>
              <div className="input-field col s6">
                <input placeholder="Category" id="category" type="text" className="validate" />
                <label className='active' htmlFor="category">Category</label>
              </div>
            </div>
            <div className="row">
              <div className="input-field col s12">
                <input disabled value="I am not editable" id="disabled" type="text" className="validate" />
                <label className='active' htmlFor="disabled">Disabled</label>
              </div>
            </div>
            <div className="row">
              <div className="input-field col s12">
                <input id="password" type="password" className="validate" />
                <label className='active' htmlFor="password">Password</label>
              </div>
            </div>
            <div className="row">
              <div className="input-field col s12">
                <input id="email" type="email" className="validate" />
                <label className='active' htmlFor="email">Email</label>
              </div>
            </div>
            <div className="row">
              <div className="col s12">
                This is an inline input field:
                <div className="input-field inline">
                  <input id="email" type="email" className="validate" />
                  <label className='active' htmlFor="email" data-error="wrong" data-success="right">Email</label>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default CategoryForm
