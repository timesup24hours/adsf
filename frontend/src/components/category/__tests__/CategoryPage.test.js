// npm packages
import configureMockStore from 'redux-mock-store'
import CategoryPageWrapper, { CategoryPage } from '../CategoryPage'
import CategoryForm from '../CategoryForm'

// create mock store
const mockStore = configureMockStore();
const menu = {
  pending: false
}
const store = mockStore({ menu: menu });

test('# CategoryPageWrapper', () => {
  const wrapper = shallow(<CategoryPageWrapper store={store} />)
  expect(wrapper).toMatchSnapshot();
})

test('# CategoryPage', () => {
  const wrapper = shallow(<CategoryPage menu={menu} />)
  expect(wrapper).toMatchSnapshot();

  const app = mount(<CategoryPage menu={menu} />)
  expect(app.find('#CategoryPage-container').exists()).toBe(true)
})

test('# CategoryPage', () => {
  const app = mount(<CategoryPage menu={{ pending: true }} />)
  expect(app.find('#CategoryPage-container').exists()).toBe(false)
})
