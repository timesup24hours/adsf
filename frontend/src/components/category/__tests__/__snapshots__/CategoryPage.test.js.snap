exports[`test # CategoryPage 1`] = `
<div
  className="CategoryPage-container"
  id="CategoryPage-container">
  <CategoryForm
    menu={
      Object {
        "pending": false,
      }
    } />
</div>
`;

exports[`test # CategoryPageWrapper 1`] = `
<Component
  menu={
    Object {
      "pending": false,
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  } />
`;
